scripts/validate-all-configs.sh

```diff
@@ -0,0 +1,303 @@
+        if baseline_passed and file in patch and not patch[file]:
+            regressions.append(file)
+
+    if regressions:
+        with open(report_file, 'w') as f:
+            f.write("REGRESSION DETECTED!\n")
+            f.write("==================\n\n")
+            f.write(f"Found {len(regressions)} config(s) that passed baseline but failed with patch:\n\n")
+            for config in sorted(regressions):
+                f.write(f"  - {config}\n")
+
+        print(f"Found {len(regressions)} regression(s)")
+        sys.exit(1)
+    else:
+        with open(report_file, 'w') as f:
+            f.write("No regressions detected.\n")
+            f.write("All configs that passed baseline also pass with patch.\n")
+
+        print("No regressions found")
+        sys.exit(0)
+
+if __name__ == "__main__":
+    main()
+EOF
+
+    if python3 compare_results.py "$baseline" "$patch" "$report"; then
+        print_status "No regressions found!"
+        rm -f compare_results.py
+        return 0
+    else
+        print_error "Regressions detected!"
+        cat "$report"
+        rm -f compare_results.py
+        return 1
+    fi
+}
+
+cleanup() {
+    if [ "$STASH_APPLIED" = "true" ]; then
+        print_status "Restoring original code state..."
+        git stash pop --quiet 2>/dev/null || true
+    fi
+}
+
+main() {
+    print_status "Starting config validation test"
+
+    trap cleanup EXIT
+
+    if ! build_evergreen_cli; then
+        print_error "Failed to build evergreen CLI"
+        exit 1
+    fi
+
+    if ! setup_evergreen_config; then
+        print_error "Failed to setup evergreen CLI configuration"
+        exit 1
+    fi
+
+    if ! download_configs; then
+        print_error "Failed to download configs"
+        exit 1
+    fi
+
+    local total_configs=$(find "$CONFIGS_DIR" -name "*.yml" -o -name "*.yaml" | wc -l)
+    print_status "Found $total_configs config files to validate"
+
+    if [ "$total_configs" -eq 0 ]; then
+        print_error "No config files found to validate"
+        exit 1
+    fi
+
+    STASH_APPLIED="false"
+    if git diff --quiet validator/ && git diff --cached --quiet validator/; then
+        print_status "No uncommitted changes to validator/, comparing against base branch"
+    else
+        print_status "Detected uncommitted changes to validator/"
+        print_status "Stashing changes for baseline validation..."
+        if ! git stash push -m "temp-stash-for-validation" validator/; then
+            print_error "Failed to stash validator changes"
+            exit 1
+        fi
+        STASH_APPLIED="true"
+    fi
+
+    print_status "Building baseline validator..."
+    if ! build_validator; then
+        if [ "$STASH_APPLIED" = "true" ]; then
+            git stash pop --quiet
+        fi
+        print_error "Failed to build baseline validator"
+        exit 1
+    fi
+
+    print_status "Running baseline validation..."
+    if ! run_validation "$BASELINE_RESULTS" true; then
```

Comment:
This is passing in `true` as a second argument to `run_validation` which `run_validation` ignores. (the same is true for line 271)